tenets/codelingo/go/break-select-in-for/example.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import "fmt"
+
+func main() {
+	// We generally write break statements to break out of repetitious things like for loops
+	fmt.Println("Expecting 0, 1:")
+	for i := range []int{0, 1, 2} {
+		fmt.Println(i)
+		if i == 1 {
+			break // Non-issue
+		}
+	}
+
+	// and don't often break from selects, so it's easy to forget that it's possible
+	fmt.Println("Expecting 0:")
+	c1 := make(chan int)
+	go func() {
```

Comment:
FWIW, this is one of the cases where buffered channels make sense and (IMO) reduce confusion by skipping the goroutine: https://play.golang.org/p/uvyrMmaOmFF

Reply:
or indeed https://play.golang.org/p/yg-B1s3KHuN